File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    dict := make(map[int]int)

    for index, num := range nums {
        if val, ok := dict[target - num]; ok{
            return []int{index, val}
        } else {
            dict[num] = index
        }
    }
    return nil
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        map = dict()
        for index, num in enumerate(nums):
            if target - num in map:
                return [index, map[target-num]]
            else:
                map[num] = index        